EIPS/eip-6160.md

```diff
@@ -0,0 +1,80 @@
+
+Optimisim and Arbitrum, optimistic L2s which are already live on Ethereum today currently have their own custom token interfaces¹ that allow their bridge contracts to mint & burn tokens on L1 & L2. In order to remove the cognitive load on token issuers & developers from extending multiple interface for all the various L2s that will live on Ethereum. It becomes necessary to define a standardized API that all token issuers should implement in order to provide these trustless bridge contracts with the interface needed to transport tokens across these different chains.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Smart contracts implementing the EIP-6160 standard MUST implement all of the functions in the `EIP-6160` interface.
+
+Smart contracts implementing the EIP-6160 standard MUST implement the `EIP-165 supportsInterface` function and MUST return the constant value `true` if *`0x01cbdea7`* is passed through the `interfaceID` argument.
+
+1. Any contract complying with [EIP-20](./eip-20.md) when extended with this EIP, **MUST** implement the following interface:
+
+```solidity
+// The EIP-165 identifier of this interface is 0xbbb8b47e
+interface IERC6160Ext20 is IERC5679Ext20, IERC_ACL_CORE {}
+```
+
+2. Any contract complying with [EIP-721](./eip-721.md) when extended with this EIP, **MUST** implement the following interface:
+
+```solidity
+// The EIP-165 identifier of this interface is 0xa75a5a72
+*interface IERC6160Ext721 is IERC5679Ext721, IERC_ACL_CORE {}
+```
+
+3. Any contract complying with [EIP-1155](./eip-1155.md) when extended with this EIP, **MUST** implement the following interface:
+
+```solidity
+// The EIP-165 identifier of this interface is 0x9f77104c
+interface IERC6160Ext1155 is IERC5679Ext1155, IERC_ACL_CORE {}
+```
+
+4. It is RECOMMENDED for compliant contracts to implement the optional extension `IERC_ACL_GENERAL`.
+5. Compliant contracts MAY implement the optional extension `IERC_ACL_METADATA`.
+
+## Rationale
+
+1. We have chosen the [EIP-5679](./eip-5679.md) standard to provide the mint & burn interface.
+2. We have chosen [EIP-5982](./eip-5982.md) standard to provide the access control interface for trustless bridge contracts to check their permissions to mint & burn `ERC20` tokens.
+3. We have chosen NOT to create new events but to require the usage of existing transfer event as required by EIP-20 EIP-721 and EIP-1155 for maximum compatibility.
+
+## Implementation
+We have made an example implementation for this EIP at [polytope-labs implementation](https://github.com/polytope-labs/multichain-native-tokens/tree/main/src/tokens)
```

Comment:
We don't allow external reference implementations. If you'd like, you can create a minimal reference implementation and include it in the `assets` directory under a permissive license (`CC0-1.0` preferred.) Failing that, just remove the whole section.